samtranslator/model/apigateway.py

```diff
@@ -153,6 +156,26 @@ def _status_code_string(self, status_code):
         return None if status_code is None else str(status_code)
 
 
+class ApiGatewayDomainName(Resource):
+    resource_type = 'AWS::ApiGateway::DomainName'
+    property_types = {
+            'RegionalCertificateArn': PropertyType(False, is_str()),
+            'DomainName': PropertyType(True, is_str()),
```

Comment:
Does `is_str()` on a lot of these properties mean intrinsics won't be supported?

Reply:
> Really excited to see movement on custom domain support! Lots of questions around edge case behavior. I don't think a single happycase test is sufficient here.
> 
> Also, are you planning to add an example as part of this PR?

I probably should add an example. Let me add that in the next rev thanks!

Reply:
Great, thanks! We've received feedback from customers that they really value the examples so we should definitely be adding them for new features.